modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue

```diff
@@ -431,37 +459,44 @@ export default {
       this.prepErrors();
       this.docReady = true;
       this.modal.triggerFocusRefresh++;
-    } else {
-      this.$nextTick(async () => {
-        await this.loadNewInstance();
-        this.evaluateConditions();
-        this.modal.triggerFocusRefresh++;
-      });
-    }
-    apos.bus.$on('content-changed', this.onContentChanged);
-  },
-  unmounted() {
-    apos.bus.$off('content-changed', this.onContentChanged);
-  },
-  methods: {
-    async saveHandler(action) {
+    },
+    async instantiateNewInstance () {
+      this.docReady = false;
+      const newInstance = await this.getNewInstance();
+      this.original = newInstance;
+      if (newInstance && newInstance.type !== this.docType) {
+        this.docType = newInstance.type;
+      }
+      this.docFields.data = newInstance;
+      const slugField = this.schema.find(field => field.name === 'slug');
+      if (slugField) {
+        // As a matter of UI implementation, we know our slug input field will
+        // automatically change the empty string to the prefix, so to
+        // prevent a false positive for this being considered a change,
+        // do it earlier when creating a new doc.
+        this.original.slug = this.original.slug || slugField.def || slugField.prefix || '';
+      }
+      this.prepErrors();
+      this.docReady = true;
+      this.evaluateConditions();
+    },
+    async saveHandler(action, saveOpts = {}) {
       this.triggerValidation = true;
-      this.$nextTick(async () => {
-        if (this.savePreference !== action) {
-          this.setSavePreference(action);
-        }
-        if (!this.errorCount) {
-          this[action]();
-        } else {
-          this.triggerValidation = false;
-          await apos.notify('apostrophe:resolveErrorsBeforeSaving', {
-            type: 'warning',
-            icon: 'alert-circle-icon',
-            dismiss: true
-          });
-          this.focusNextError();
-        }
-      });
+      await this.$nextTick();
+      if (this.savePreference !== action) {
+        this.setSavePreference(action);
+      }
+      if (!this.errorCount) {
+        await this[action](saveOpts);
+      } else {
+        this.triggerValidation = false;
+        await apos.notify('apostrophe:resolveErrorsBeforeSaving', {
+          type: 'warning',
+          icon: 'alert-circle-icon',
+          dismiss: true
+        });
+        this.focusNextError();
+      }
```

Comment:
I updated this `saveHandler` to make it awaitable. otherwise when saving and switching locale and switch happened before the save...